### PR TITLE
add git commit hash in cargo-contract version

### PR DESCRIPTION
**Pull request by atenjin**

### CODE REVIEW

**Review by cmichi — CHANGES_REQUESTED** [review]
@atenjin The commit hash is missing in the second command here:
```
$ cargo run -- --version
cargo-contract 0.8.0-bd0db50-x86_64-linux-gnu
$ cargo run -- contract --version
cargo-contract 0.8.0
```

Besides that I think it's fine the way you copied the `generate_cargo_keys` stuff, avoiding polluting the codebase with substrate-named environment variables. So you can remove the comments regarding that.

File: build.rs [hunk 1]
```diff
@@ -15,18 +15,23 @@
...
     ffi::OsStr,
     fs::File,
     io::{prelude::*, Write},
     iter::Iterator,
     path::{Path, PathBuf},
+    process::Command,
 };
 
 use anyhow::Result;
+use platforms::*;
```
Comment (cmichi) [inline build.rs:29]: I would prefer not having a wildcard import here.
```suggestion
use platforms::{TARGET_ARCH, TARGET_ENV, TARGET_OS};
```

Comment (atenjin) [reply build.rs]: have fixed

**Comment by atenjin** [comment]
> @atenjin The commit hash is missing in the second command here:
> 
> ```
> $ cargo run -- --version
> cargo-contract 0.8.0-bd0db50-x86_64-linux-gnu
> $ cargo run -- contract --version
> cargo-contract 0.8.0
> ```
> 
> Besides that I think it's fine the way you copied the `generate_cargo_keys` stuff, avoiding polluting the codebase with substrate-named environment variables. So you can remove the comments regarding that.

thanks, I find out the place to add version for this tip. 

Now I have removed the todo command and added suitable version hint.

**Review by cmichi — APPROVED** [review]
Thanks @atenjin !

**Pull request merged**